--- python-evas-0.7.3/evas.orig/evas.c_evas_canvas.pxi	2010-10-01 01:03:46.000000000 +0200
+++ python-evas-0.7.3/evas/evas.c_evas_canvas.pxi	2011-12-16 20:42:13.254098609 +0100
@@ -117,7 +117,7 @@
            obscured_clear, render_updates, render, norender, freeze_get,
            output_method_set, output_method_get, output_method
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.obj = NULL
         self._callbacks = [None] * evas_canvas_event_callbacks_len
 
--- python-evas-0.7.3/evas.orig/evas.c_evas_object.pxi	2010-10-01 01:03:46.000000000 +0200
+++ python-evas-0.7.3/evas/evas.c_evas_object.pxi	2011-12-16 20:42:25.217432342 +0100
@@ -217,7 +217,7 @@
        anti_alias_set, anti_alias_get, anti_alias, pointer_mode_set,
        pointer_mode_get, pointer_mode
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.obj = NULL
         self.evas = None
         self.data = dict()
--- python-evas-0.7.3/evas.orig/evas.c_evas_object_smart.pxi	2010-10-01 01:03:46.000000000 +0200
+++ python-evas-0.7.3/evas/evas.c_evas_object_smart.pxi	2011-12-16 20:45:23.880771658 +0100
@@ -359,7 +359,7 @@
 
     @see: L{ClippedSmartObject}
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._smart_callbacks = dict()
         cls = self.__class__
         self._m_delete = _smart_class_get_impl_method(cls, "delete")
@@ -501,7 +501,7 @@
                 break
         else:
             raise ValueError("Callback %s was not registered with event %r" %
-                             (func, e))
+                             (func, event))
 
         lst.pop(i)
         if lst:
